src/archive.rs

```diff
@@ -241,7 +275,11 @@ impl<'a> EntriesFields<'a> {
         loop {
             // Seek to the start of the next header in the archive
             let delta = self.next - self.archive.inner.pos.get();
-            self.archive.skip(delta)?;
+            if let Some(seekable_archive) = self.seekable_archive {
+                seekable_archive.skip_with_seek(delta)?;
```

Comment:
Coudld the `skip_with_seek` method get folded into a helper method of `Entries` to internally dispatch to a read-the-bytes or call-the-`seek` method? I don't think that `Archive::skip` needs to be called anywhere else and I think it'd be good to centralize skipping logic.